simpleclient_hibernate/src/main/java/io/prometheus/client/hibernate/HibernateStatisticsCollector.java

```diff
@@ -448,25 +473,182 @@ public double getValue(Statistics statistics) {
     );
   }
 
+  private List<MetricFamilySamples> getPerQueryMetrics() {
+    List<MetricFamilySamples> metrics = new ArrayList<MetricFamilySamples>();
+
+    metrics.addAll(Arrays.asList(
+
+        createCounterForQuery("hibernate_per_query_cache_hit_total",
+                "Global number of cache hits for query (getCacheHitCount)",
+            new ValueProviderPerQuery() {
+              @Override
+              public double getValue(Statistics statistics, String query) {
+                return statistics.getQueryStatistics(query)
+                        .getCacheHitCount();
+              }
+            }
+        ),
+        createCounterForQuery("hibernate_per_query_cache_miss_total",
+                "Global number of cache misses for query (getCacheMissCount)",
+            new ValueProviderPerQuery() {
+              @Override
+              public double getValue(Statistics statistics, String query) {
+                return statistics.getQueryStatistics(query)
+                        .getCacheMissCount();
+              }
+            }
+        ),
+        createCounterForQuery("hibernate_per_query_cache_put_total",
+                "Global number of cache puts for query (getCachePutCount)",
+            new ValueProviderPerQuery() {
+              @Override
+              public double getValue(Statistics statistics, String query) {
+                return statistics.getQueryStatistics(query)
+                        .getCachePutCount();
+              }
+            }
+        ),
+        createCounterForQuery("hibernate_per_query_execution_total",
+                "Global number of executions for query (getExecutionCount)",
+            new ValueProviderPerQuery() {
+              @Override
+              public double getValue(Statistics statistics, String query) {
+                return statistics.getQueryStatistics(query)
+                        .getExecutionCount();
+              }
+            }
+        ),
+        createCounterForQuery("hibernate_per_query_execution_rows_total",
+                "Global number of rows for all executions of query (getExecutionRowCount)",
+            new ValueProviderPerQuery() {
+              @Override
+              public double getValue(Statistics statistics, String query) {
+                return statistics.getQueryStatistics(query)
+                        .getExecutionRowCount();
+              }
+            }
+        ),
+        createGaugeForQuery("hibernate_per_query_execution_min_time",
+                "Minimum execution time of query in milliseconds (getExecutionMinTime)",
```

Comment:
Seconds are the base unit, and the metric name should also include the unit

Reply:
Added "_ms" to all three gauge metrics.

Reply:
You should convert to seconds, as that's standard in Prometheus

Reply:
@brian-brazil milliseconds are being converted to seconds now. Also added "seconds" to the metric names.